src/parent/reducers/reducers.js

```diff
@@ -1,12 +1,61 @@
 import { combineReducers } from 'redux';
+import reduceReducers from 'reduce-reducers';
+
+import { PARENT, SIDEBAR } from '../../shared/constants/actionTypes';
 
 import childWindows from './childWindows';
+import closedWindows from './closedWindows';
 import dragOut from './dragOut';
 
-const rootReducer = combineReducers({
-    childWindows,
-    dragOut
-});
+const rootReducer = reduceReducers(
```

Comment:
Another idea could be to structure the state shape like this:

```js
{
    childWindows: {
        window0001: { ... },
        window0002: { ... }
    },
    openWindows: ['window0001'], // (window IDs)
    closedWindow: ['window0002'], // (window IDs)
    ...
}
```

Might not be worth doing this now though, as I expect it would be a fairly large change.